langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockChatModel.java

```
@@ -107,27 +103,15 @@ public BedrockChatModel(Builder builder) {
                 : builder.client;
-        this.defaultRequestParameters = ChatRequestParameters.builder()
-                .modelName(getOrDefault(
-                        modelId,
-                        nonNull(builder.defaultRequestParameters)
-                                ? builder.defaultRequestParameters.modelName()
-                                : null))
-                .temperature(
-                        nonNull(builder.defaultRequestParameters)
-                                ? builder.defaultRequestParameters.temperature()
-                                : null)
-                .maxOutputTokens(getOrDefault(
-                        builder.maxTokens,
-                        nonNull(builder.defaultRequestParameters)
-                                ? builder.defaultRequestParameters.maxOutputTokens()
-                                : null))
-                .topP(nonNull(builder.defaultRequestParameters) ? builder.defaultRequestParameters.topP() : null)
-                .stopSequences(copyIfNotNull(
-                        nonNull(builder.defaultRequestParameters)
-                                ? builder.defaultRequestParameters.stopSequences()
-                                : null))
-                .build();
+        this.defaultRequestParameters = isNull(builder.defaultRequestParameters)
+                ? ChatRequestParameters.builder().modelName(this.modelId).build()
+                : ChatRequestParameters.builder()
+                        .modelName(this.modelId)
+                        .temperature(builder.defaultRequestParameters.temperature())
+                        .maxOutputTokens(builder.defaultRequestParameters.maxOutputTokens())
+                        .topP(builder.defaultRequestParameters.topP())
+                        .stopSequences(copyIfNotNull(builder.defaultRequestParameters.stopSequences()))
+                        .build();
 
         validate(this.defaultRequestParameters);
```

this will validate paramets after copying only supported ones from `builder.defaultRequestParameters`, so it means that if some unsupported parameter is passed in the `builder.defaultRequestParameters`, it will be silently ignored